File: app/src/main/java/com/jon/cotgenerator/utils/Notify.java
package com.jon.cotgenerator.utils;

import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.snackbar.Snackbar;

public final class Notify {
    private Notify() {
    }

    private static void snackbar(View root, String str, int backgroundColour, int textColour) {
        Snackbar snackbar = Snackbar.make(root, str, Snackbar.LENGTH_LONG);
        View view = snackbar.getView();
        view.setBackgroundColor(backgroundColour);
        TextView text = view.findViewById(com.google.android.material.R.id.snackbar_text);
        text.setTextColor(textColour);
        snackbar.show();
    }

    public static void green(View root, String str) {
        snackbar(root, str, Color.GREEN, Color.BLACK);
    }

    public static void red(View root, String str) {
        snackbar(root, str, Color.RED, Color.BLACK);
    }

    public static void blue(View root, String str) {
        snackbar(root, str, Color.BLUE, Color.WHITE);
    }

    public static void orange(View root, String str) {
        snackbar(root, str, Color.parseColor("#FFA600"), Color.BLACK);
    }
}

File: common/src/main/java/com/jon/common/utils/Notify.kt
package com.jon.common.utils

import android.content.Context
import android.graphics.Color
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.google.android.material.R
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.google.android.material.snackbar.Snackbar
import timber.log.Timber

object Notify {
    private const val DEFAULT_TEXT = "OK"
    private val DEFAULT_ACTION = View.OnClickListener { /* do nothing */ }

    private var anchor: View? = null

    fun setAnchor(view: View) {
        this.anchor = view
    }

    private fun snackbar(
            root: View,
            message: String,
            backgroundColour: Int,
            textColour: Int,
            action: View.OnClickListener?,
            actionMsg: String?
    ) {
        val snackbar = Snackbar.make(root, message, Snackbar.LENGTH_LONG)
        snackbar.view.setBackgroundColor(backgroundColour)
        val text = snackbar.view.findViewById<TextView>(R.id.snackbar_text)
        text.setTextColor(textColour)
        text.maxLines = 20 // don't crop off any longer messages

        anchor?.let { snackbar.anchorView = it }

        /* Add any onclick actions */
        if (action != null && actionMsg != null) {
            snackbar.setAction(actionMsg, action)
            snackbar.setActionTextColor(textColour)
        }
        snackbar.show()
    }

    fun red(
            root: View,
            message: String,
            action: View.OnClickListener = DEFAULT_ACTION,
            actionMsg: String = DEFAULT_TEXT
    ) {
        Timber.e(message)
        snackbar(root, message, Color.RED, Color.BLACK, action, actionMsg)
    }

    fun green(
            root: View,
            message: String,
            action: View.OnClickListener = DEFAULT_ACTION,
            actionMsg: String = DEFAULT_TEXT
    ) {
        Timber.i(message)
        snackbar(root, message, Color.GREEN, Color.BLACK, action, actionMsg)
    }

    fun orange(
            root: View,
            message: String,
            action: View.OnClickListener = DEFAULT_ACTION,
            actionMsg: String = DEFAULT_TEXT
    ) {
        Timber.w(message)
        val orange = Color.parseColor("#FFA600")
        snackbar(root, message, orange, Color.BLACK, action, actionMsg)
    }

    fun blue(
            root: View,
            message: String,
            action: View.OnClickListener = DEFAULT_ACTION,
            actionMsg: String = DEFAULT_TEXT
    ) {
        Timber.i(message)
        snackbar(root, message, Color.BLUE, Color.WHITE, action, actionMsg)
    }

    fun yellow(
            root: View,
            message: String,
            action: View.OnClickListener = DEFAULT_ACTION,
            actionMsg: String = DEFAULT_TEXT
    ) {
        Timber.i(message)
        snackbar(root, message, Color.YELLOW, Color.BLACK, action, actionMsg)
    }

    fun toast(context: Context, message: String) {
        Timber.i(message)
        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
    }

    fun alert(context: Context, message: String) {
        MaterialAlertDialogBuilder(context)
                .setTitle("Alert")
                .setMessage(message)
                .setPositiveButton(android.R.string.ok, null)
                .show()
    }
}